src/upchunk.spec.ts

```diff
@@ -282,7 +282,7 @@ test('progress event fires the correct upload percentage', (done) => {
         done('All scopes not completed');
       }
     });
-    expect(progressCallback).toHaveBeenCalledTimes(7);
```

Comment:
Since the upload endpoints are mocked out, it will reliably be 4 here in the test env, right?

Reply:
That's correct -- previously, there were two different places in the code dispatching the `progress` events, and it was awkward to follow the logic.  I simplified that down to one location, hence the change in the number of `progressCallback`s.